web3/inpage.js

```diff
@@ -151,10 +151,33 @@ class Web3FrameProvider {
 
 if (!window.chrome || !window.chrome.extension) {
   console.log('Parity - Injecting Web3');
-  window.web3 = {
-    currentProvider: new Web3FrameProvider()
+
+  const web3 = {
+    currentProvider: new Web3FrameProvider(),
+    injectedWeb3: null
   };
 
+  const proxiedWeb3 = new Proxy(web3, {
```

Comment:
Yes, I'll look into that. However I don't know how the dapp would handle to fact that we don't return the web3 instance right away. (ie. it would need to make a second request to finally get it ; which could be the case in most of the dapps, but I'm not sure)